homeassistant/components/versasense/sensor.py

```diff
@@ -0,0 +1,74 @@
+"""Support for VersaSense sensor peripheral."""
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the sensor platform."""
+    consumer = hass.data[DOMAIN]['consumer']
+    peripheral = hass.data[DOMAIN][discovery_info['identifier']]
+    parent_name = discovery_info['parentName']
+    unit = discovery_info['unit']
+    measurement = discovery_info['measurement']
+
+    add_devices(
+        [VSensor(peripheral, parent_name, unit, measurement, consumer)]
+    )
+
+
+class VSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, peripheral, parent_name, unit, measurement, consumer):
+        """Initialize the sensor."""
+        self._state = None
+        self._name = parent_name + ": " + measurement
+        self._parent_mac = peripheral.parentMac
+        self._identifier = peripheral.identifier
+        self._unit = unit
+        self._measurement = measurement
+        self.consumer = consumer
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def parent_mac(self):
+        """Return the parent mac address of the sensor."""
+        return self._parent_mac
```

Comment:
Had to use the parent mac address, identifier and measurement of the sensor/actuator to create a fully unique id. This because a parent device can have multiple peripherals (identifier) with multiple measurements. Using only the parent mac address created problems because the id was already in use.